src/cascadia/TerminalSettingsModel/EnumMappings.cpp

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "ActionAndArgs.h"
+#include "JsonUtils.h"
+#include "TerminalSettingsSerializationHelpers.h"
+
+#include "EnumMappings.h"
+#include "EnumMappings.g.cpp"
+
+using namespace winrt;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace ::Microsoft::Terminal::Settings::Model;
+
+namespace winrt::Microsoft::Terminal::Settings::Model::implementation
+{
+    IMap<hstring, winrt::Windows::UI::Xaml::ElementTheme> EnumMappings::ElementTheme()
+    {
+        auto map = single_threaded_map<hstring, winrt::Windows::UI::Xaml::ElementTheme>();

```

Comment:
Am I reading this right, we're building the map of string -> value _every time this is called_? Should we instead be doing one of those "magic statics" things that Dustin loves that only builds the map the first time this function is called, and then just keeps it for future callers?

Reply:
Yup you're right, we should be caching the map the first time it's called. Is that magic static thing called a function level static variable perhaps?

Reply:
there's a bunch of these scattered throughout the codebase, not totally sure what its formal name is 😛 

```c++
        // use C++11 magic statics to make sure we only do this once.
        static bool initialized = []() { ... }();
```

Reply:
ooh i could also throw this function in a macro

Reply:
that looks insane, does that just set to `initialized` to true the first time it's called then we just check `if(initialized)`

Reply:
Not even that, actually. It's a big trick. The bool was to initialize something whose value we didn't care about. Because we care about the expression result here, we can save time/mental overhead by static-fying that value.

You'll more likely want a version of this:

```
static auto map = [](){
    auto map = single_threaded_map<T>{};
    ///// do stuff to map
    return map;
}()
return map;
```